pkg/trace/agent/agent.go

```diff
@@ -422,25 +422,25 @@ func (a *Agent) processStats(in pb.ClientStatsPayload, lang, tracerVersion strin
 	for i, group := range in.Stats {
 		n := 0
 		for _, b := range group.Stats {
-			normalizeStatsGroup(&b, lang)
-			if !a.Blacklister.AllowsStat(&b) {
+			normalizeStatsGroup(b, lang)
+			if !a.Blacklister.AllowsStat(b) {
 				continue
 			}
-			a.obfuscateStatsGroup(&b)
-			a.Replacer.ReplaceStatsGroup(&b)
+			a.obfuscateStatsGroup(b)
+			a.Replacer.ReplaceStatsGroup(b)
 			group.Stats[n] = b
 			n++
 		}
 		in.Stats[i].Stats = group.Stats[:n]
-		mergeDuplicates(in.Stats[i])
+		mergeDuplicates(*in.Stats[i])
 	}
 	return in
 }
 
 func mergeDuplicates(s pb.ClientStatsBucket) {
 	indexes := make(map[stats.Aggregation]int, len(s.Stats))
 	for i, g := range s.Stats {
-		a := stats.NewAggregationFromGroup(g)
+		a := stats.NewAggregationFromGroup(*g)
```

Comment:
Would it be safe to just change `stats.NewAggregationFromGroup` to take a pointer instead?